### ISSUE TITLE
Required parameters are not checked in JS files 

### ISSUE THREAD

**Issue by que-etc** [Microsoft/TypeScript#22080]
**TypeScript Version:**  2.8.0-dev.20180220

**Search Terms:** 
* jsdoc
* checkJs
* function type
* arguments

**Code**

```js
/** @type {function(string): void} */
const func = (value) => {
    // ...
}

// Following line should raise an error as 'func' invocation
// is missing required parameter, but it doesn't happen.
func();
```

**Expected behavior:**
An error `Expected 1 argument, but got 0.` should be thrown.

**Actual behavior:**
No error is thrown. TypeScript allows to invoke former function without passing arguments to it.

**Comment by andy-ms** [comment]
Probably due to #13905. @mhegazy

**Comment by mhegazy** [comment]
I think it is rather due to https://github.com/Microsoft/TypeScript/pull/13905

**Comment by sandersn** [comment]
Fix is up at #22692
